client/src/mac.rs

```diff
@@ -0,0 +1,41 @@
+use serde::{Deserialize, Serialize};
+#[cfg(feature = "native_crypto")]
+use subtle::ConstantTimeEq;
+use zbus::zvariant::Type;
+
+#[derive(Deserialize, Serialize, Type, Debug, Clone)]
+pub struct Mac(Vec<u8>);
+
+impl Mac {
+    pub fn new(inner: Vec<u8>) -> Self {
```

Comment:
Is this needed? If yes, make it a From impl

Reply:
I made it `pub(crate)`. We can use `Mac(bytes)` if you prefer. 

The wart I mentioned is that this is not intended to be constructed by the user (since it requires length verification), therefore it needs a public method to be compared against a slice.